monkey-patch/defaultnormal_vertex.glsl.js

```diff
@@ -34,6 +34,42 @@ module.exports = [
 
 	"#endif",
 
-"#endif"
+"#endif",
+
+"#ifdef USE_TANGENT",
+
+	"#ifdef FLIP_SIDED",
+
+		"objectTangent = -objectTangent;",
+
+	"#endif",
+
+	"#ifndef INSTANCE_TRANSFORM",
+
+		"vec3 transformedTangent = normalMatrix * objectTangent;",
+
+	"#else",
+
+		"#ifndef INSTANCE_MATRIX",
+	
+			"mat4 _instanceMatrix = getInstanceMatrix();",
+	
+			"#define INSTANCE_MATRIX",
+	
+		"#endif",
+	
+		"#ifndef INSTANCE_UNIFORM",
+	
+			"vec3 transformedTangent =  transposeMat3( inverse( mat3( modelViewMatrix * _instanceMatrix ) ) ) * objectTangent;",
```

Comment:
I think this though, shouldn't be another computation, for both of this and the `#else` below, this matrix should always be available, so we should cache `transposeMat3( inverse( mat3( modelViewMatrix * _instanceMatrix ) ) )` above when the normal is computed.

Reply:
You're absolutely right - I should have realized this can be refactored significantly.
PR updated.